src/components/text-input/TextInput.root.tsx

```diff
@@ -74,6 +92,34 @@ const TextInputRoot = styled.div<PTextInputRoot>(
                     horizontal: TEXT_INPUT_VALUES_MAPPING[size].spacing,
                 })
             )};
+
+            span {
+                position: absolute;
+                white-space: nowrap;
+                background-color: ${hasValue ? backgroundColor : 'transparent'};
+                color: ${colors.text};
+                opacity: 1;
+                transform: translate(${hasValue ? LABEL_POSITIONS[size] : '0, 0'})
+                    scale(${hasValue ? '0.7' : '1'});
+                ${applyShape({ radius: 4 })};
+                ${applyPadding(Spacing.symmetric({ vertical: 25, horizontal: 75 }))};
+                ${applyMargin(
+                    Spacing.only(
+                        'left',
+                        leadingIcon === 'none' ? 0 : TEXT_INPUT_VALUES_MAPPING[size].labelMargin
+                    )
+                )};
+                ${applyTextStyles({
+                    size: TEXT_INPUT_VALUES_MAPPING[size].labelSize,
+                    inheritLineHeight: true,
+                })};
+
+                transition: all ${animation.fastest}ms linear;
+            }
+
+            .clear {
```

Comment:
the trailing icon will have different options (basically just running a callback function, or be decorative). Either way the naming is a bit awkward, when you somehow imply a function to it, which it can, but not has to be doing. 

e.g. it could also be a trigger to perform a search.

Better naming would be `.trailing-icon` and instead of `isClearable` just name it `isTrailingIconFunctional`? Open to suggestions!